TypeScript errors in 4.0.0-beta.4

It seems that some of the typings provided in 4.0.0-beta.4 are invalid/incomplete.

cc: @bartocc 

## Version

4.0.0-beta.4

## Expected Behavior

No type errors on application build

## Actual Behavior

```
../node_modules/ember-changeset/index.d.ts:7:26 - error TS2499: An interface can only extend an identifier/qualified-name with optional type arguments.

7 interface Config extends BufferedChangesetConstructorParameters[3] {
                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

../node_modules/ember-changeset/index.d.ts:19:17 - error TS7010: 'changeset', which lacks return-type annotation, implicitly has an 'any' return type.

19 export function changeset(...args: changesetFunctionsParameters);
                   ~~~~~~~~~

../node_modules/ember-changeset/index.d.ts:20:17 - error TS7010: 'Changeset', which lacks return-type annotation, implicitly has an 'any' return type.

20 export function Changeset(...args: changesetFunctionsParameters);
                   ~~~~~~~~~
```

--- a/index.d.ts
+++ b/index.d.ts
@@ -1,12 +1,8 @@
-import { BufferedChangeset } from "validated-changeset";
+import { BufferedChangeset } from 'validated-changeset';
 
-type BufferedChangesetConstructorParameters = ConstructorParameters<
-  typeof BufferedChangeset
->;
+type BufferedChangesetConstructorParameters = ConstructorParameters<typeof BufferedChangeset>;
 
-interface Config extends BufferedChangesetConstructorParameters[3] {
-  changeset?: typeof EmberChangeset;
-}
+type Config = BufferedChangesetConstructorParameters[3] & { changeset?: typeof EmberChangeset; }
 
 type changesetFunctionsParameters = [
   BufferedChangesetConstructorParameters[0],
@@ -16,5 +12,5 @@ type changesetFunctionsParameters = [
 ];
 
 export class EmberChangeset extends BufferedChangeset {}
-export function changeset(...args: changesetFunctionsParameters);
-export function Changeset(...args: changesetFunctionsParameters);
+export function changeset(...args: changesetFunctionsParameters): EmberChangeset;
+export function Changeset(...args: changesetFunctionsParameters): EmberChangeset;

